src/core/client.ts

```diff
@@ -787,67 +838,124 @@ export class Client<P = Indexable> implements Observable<ClientEvent<P>> {
         }
       });
 
+      const changedPeers: Record<
+        DocumentKey,
+        Array<{ clientID: ActorID; presence: P }>
+      > = {};
+      for (const key of keys) {
+        changedPeers[key] = this.getPeersByDocKey(key);
+      }
       this.eventStreamObserver.next({
         type: ClientEventType.PeersChanged,
-        value: keys.reduce(this.getPeersWithDocKey, {}),
+        value: {
+          type: 'initialization',
+          changedPeers,
+          peers: this.getPeers(),
+        },
       });
       return;
     }
 
     const pbWatchEvent = resp.getEvent()!;
-    const respKeys = pbWatchEvent.getDocumentKeysList();
+    const respDocKeys = pbWatchEvent.getDocumentKeysList();
+    // TODO(chacha912): Change the docKey of DocEvent to a single key value
+    // instead of an array
```

Comment:
Is `pbWatchEvent` implemented on the server to only work for a document at a time? 🤔
It would be nice to fix it on the server first.

Reply:
Since the `pbWatchEvent` is sent to each document via `subscriptionsMapByDocKey` on the server, it only applies to one document at a time. The task of [filtering document key](https://github.com/yorkie-team/yorkie/pull/463) is related and should be fixed on the server first.

Reply:
We could simplify the API a bit more. Thank you for your confirmation.